pkg/app/piped/planner/kubernetes/pipeline.go

```diff
@@ -114,6 +115,31 @@ func buildProgressivePipeline(pp *config.DeploymentPipeline, autoRollback bool,
 			CreatedAt:  now.Unix(),
 			UpdatedAt:  now.Unix(),
 		})
+
+		// Add a stage for rolling back script run stages.
+		for i, s := range pp.Stages {
+			if s.Name == model.StageScriptRun {
+				// Use metadata as a way to pass parameters to the stage.
+				envStr, _ := json.Marshal(s.ScriptRunStageOptions.Env)
+				metadata := map[string]string{
+					"baseStageID": out[i].Id,
+					"onRollback":  s.ScriptRunStageOptions.OnRollback,
+					"env":         string(envStr),
+				}
```

Comment:
📝 I use metadata for the stage to recognize below on executing it.
- The commands which are executed on the rollback script run stage. (onRollback, env)
- The stage IDs of script run stages, which are used to check whether to rollback it. (baseStageID)
    - It is because I want to execute the command to rollback SCRIPT_RUN to the point where the deployment was canceled or failed.